client/components/domains/use-my-domain/utilities/get-option-info.js

```diff
@@ -172,6 +173,33 @@ export function getOptionInfo( {
 			onSelect: onConnect,
 			pricing: mappingPricing,
 		};
+
+		// We currently aren't handling ownership verification for mapped domains during sign-up
+		// See https://github.com/Automattic/nomado-issues/issues/136 for more context
+		if ( availability.ownership_verification_type !== 'no_verification_required' && isSignupStep ) {
```

Comment:
I think we should do this not only when isSignupStep is true but also if it's on a site that doesn't have a plan already. Otherwise if you already have the site and go and try to add the mapping it sends you to the auth code verification but it does not check the auth code at all - it just sends you to checkout

Reply:
Good catch! I fixed that now